src/util/inkscape.js

```diff
@@ -0,0 +1,221 @@
+const fs = require('fs')
+const path = require('path')
+const childProcess = require('child_process')
+const parallel = require('run-parallel')
+const series = require('run-series')
+const uuid = require('uuid/v4')
+const os = require('os')
+const PNG = require('pngjs').PNG
+const semver = require('semver')
+const jsdom = require('jsdom')
+const { JSDOM } = jsdom
+const tinycolor = require('tinycolor2')
+
+const PATH_TO_BUILD = path.join(os.tmpdir(), 'orca-build')
+try {
+  fs.mkdirSync(PATH_TO_BUILD)
+} catch (e) {}
+
+const fillUrl = /(^|; )fill: url\('#([^']*)'\);/
+const fillRgbaColor = /(^|; )fill: rgba\(([^)]*)\);/
+const fillOpacityZero = /(^|\s*)fill-opacity: 0;/
+const strokeOpacityZero = /(^|; )stroke-opacity: 0;/
+const opacityZero = /(^|; )opacity: 0;/
+
+/** Node wrapper for Inkscape
+ *
+ * $ apt-get install inkscape
+ * ... or on OS X:
+ * $ brew install inkscape
+ *
+ * See:
+ * - https://linux.die.net/man/1/inkscape
+ * - https://en.wikipedia.org/wiki/Inkscape
+ */
+class Inkscape {
+  constructor (pathToInkscape) {
+    this.cmdBase = pathToInkscape || 'inkscape'
+  }
+
+  /** Convert SVG to EMF
+   *
+   * @param {buffer} svg : svg data buffer
+   * @param {object} opts
+   *  - id {string}
+   * @param {function} cb
+   *  - err {null || error}
+   *  - result {buffer}
+   */
+  svg2emf (svg, opts, cb) {
+    const id = opts.id || uuid()
+    const inPath = path.join(PATH_TO_BUILD, id + '-svg')
+    const outPath = path.join(PATH_TO_BUILD, id + '-emf')
+    const cmd = `${this.cmdBase} --file ${inPath} --export-emf ${outPath}`
+
+    const destroyTmpFiles = (cb) => parallel([
+      (cb) => fs.unlink(inPath, cb),
+      (cb) => fs.unlink(outPath, cb)
+    ], cb)
+
+    // Get background color from figure's definition
+    var bgColor
+    if (opts.figure.layout.paper_bgcolor) {
```

Comment:
`figure.layout` might not always be defined (unless orca is doing something hacky I don't remember), so I would prefer:

```js
if ((opts.figure.layout || {}).paper_bgcolor) { /* */ }
```

to stay safe.

Reply:
Change made in commit e4ab53d